pandas/tests/io/sas/test_sas7bdat.py

```diff
@@ -333,3 +333,36 @@ def test_null_date(datapath):
         },
     )
     tm.assert_frame_equal(df, expected)
+
+
+@pytest.mark.parametrize("test_file", ["test2.sas7bdat", "test3.sas7bdat"])
+def test_exception_propagation_rdc_rle_decompress(datapath, monkeypatch, test_file):
+    """Errors in RLE/RDC decompression should propagate the same error."""
+    orig_np_zeros = np.zeros
+
+    def _patched_zeros(size, dtype):
```

Comment:
Is it possible to test this without patching?

e.g. maybe import `Parser` from the cython code and call `decompress` with a string `result_length`?

Reply:
Will try to come up with something

Reply:
Unfortunately I was unable to come up with any alternatives. We want to be able to test the entire call chain up to `read()` so we need to call into that. Besides, it's very complicated to set up a `Parser` instance that triggers an exception in a specific case without monkeypatching or a handcrafted broken SAS7BDAT file.

Reply:
Okay makes sense. Just to be safe, maybe the function signature should also accept `*args, **kwargs` so future uses don't break the monkeypatch.

Reply:
Ok, I will make a PR with modifications to this very piece of code soon anyway, maybe can do it then. (Have a 20x parser speed up that I’m PRing piece by piece)